labelbox_dev/pagination.py

```diff
@@ -0,0 +1,135 @@
+            for entity in page:
+                self._entities_from_last_page.append(
+                    self.deserialize_entity(entity))
+            return next(self)
+
+
+class IdentifierPaginator(BasePaginator):
+    """
+    Fetches entities by a list of identifiers by breaking them into pages of size `limit`.
+
+    Args:
+        resource: HTTP endpoint path
+        entity_class: Subclass of `Entity` to cast objects from endpoint into
+        identifiers: List of `ids`, `global_keys` or other keys to fetch entities by
+        identifiers_key: Name of identifier field e.g. `ids` or `global_keys`
+        limit: Size of pages
+    """
+
+    def __init__(self,
+                 resource,
+                 entity_class,
+                 identifiers,
+                 params=None,
+                 limit=DEFAULT_LIMIT,
+                 identifiers_key=IDS_KEY):
+        self.resource: str = resource
+        self.entity_class: Type[Entity] = entity_class
+        self.identifiers: List[str] = identifiers
+        self.identifiers_key: str = identifiers_key
+        self.limit: int = limit
+        self._current_idx = 0
+        super().__init__(resource, entity_class, params, limit)
+
+    def get_next_page(self):
+        end_idx = self._current_idx + self.limit
+        ids = self.identifiers[self._current_idx:end_idx]
+        self._current_idx += self.limit
+        if self._current_idx >= len(self.identifiers):
+            self._is_last_page = True
+
+        ids_str = ','.join(ids)
+        params = self.params.copy()
+        params.update({self.identifiers_key: ids_str})
+        response = self._session.get_request(self.resource, params)
+        return response
+
+
+class CursorPaginator(BasePaginator):
+    """
+    Loops over all pages of an HTTP endpoint and returns entities. 
+
+    Args:
+        resource: HTTP endpoint path
+        entity_class: Subclass of `Entity` to cast objects from endpoint into
+        identifiers: List of `ids`, `global_keys` or other keys to fetch entities by
+        identifiers_key: Name of identifier field e.g. `ids` or `global_keys`
+        limit: Size of pages
+    """
+
+    def __init__(self,
+                 resource,
+                 entity_class,
+                 params=None,
+                 limit=DEFAULT_LIMIT):
+        super().__init__(resource, entity_class, params, limit)
+        self._cursor = None
+        self.params.update({LIMIT_KEY: limit})
+
+    def get_next_page(self):
+        params = self.params.copy()
+        if self._cursor:
+            params.update({CURSOR_KEY: self._cursor})
+
+        response = self._session.get_request(self.resource, params)
+        self._cursor = response.get(NEXT_CURSOR_KEY)
+        if self._cursor is None:
+            self._is_last_page = True
+
+        try:
+            self._cursor = response['next']
+            return response['data']
+        except KeyError:
+            raise ServiceError(
```

Comment:
This error sounds pretty vague, so I don't think it's worth creating a new error type for this. We should try to handle most of the errors accordingly within the `http_request` method in Session. If for some reason, `data` does not exist in the response, we can just throw a generic `LabelboxError`, or even `KeyError` with a more detailed error message.

Also, looks like we're already checking for `next` within the response in line 126, so we don't need line 131.